src/server/api/routes/channels-router.js

```diff
@@ -0,0 +1,213 @@
+const express = require('express');
+
+const router = express.Router({ mergeParams: true });
+
+// controllers
+const channelsController = require('../controllers/channels-controller');
+
+/**
+ * @swagger
+ * /modules:
+ *  post:
+ *    summary: Create a channel
+ *    description:
+ *      Will create a channel.
+ *    produces: application/json
+ *    parameters:
+ *      - in: body
+ *        name: channel
+ *        description: The channel to create.
+ *        schema:
+ *          type: object
+ *          required:
+ *              - created_at
+ *              - updated_at
+ *          properties:
+ *            title:
+ *              type: string
+ *            created_at:
+ *              type: string
+ *              format: date-time
+ *
+ *            updated_at:
+ *              type: string
+ *              format: date-time
+ *
+ *            deleted_at:
+ *              type: string
+ *              format: date-time
+ *
+ *    responses:
+ *      201:
+ *        description: Channel created
+ *      5XX:
+ *        description: Unexpected error.
+ */
+router.post('/', (req, res) => {
+  channelsController
+    .createChannel(req.body)
+    .then((result) => res.json(result))
+    .catch((error) => {
+      console.log(error);
+
+      res
+        .status(400)
+        .send('Bad request')
+        .end();
+    });
+});
+
+/**
+ * @swagger
+ * /channels:
+ *  get:
+ *    summary: Get all channels sorted alphabetically
+ *    description:
+ *      Will return all channels.
+ *    produces: application/json
+ *    responses:
+ *      200:
+ *        description: Successful request
+ *      5XX:
+ *        description: Unexpected error.
+ */
+router.get('/', (req, res, next) => {
+  channelsController
+    .getChannels()
+    .then((result) => res.json(result))
+    .catch(next);
+});
+
+/**
+ * @swagger
+ * /channels:
+ *  get:
+ *    summary: Get channels, sorted with the most recently created first
+ *    description:
+ *      Will return id and title of the channels.
+ *    produces: application/json
+ *    responses:
+ *      200:
+ *        description: Successful request
+ *      5XX:
+ *        description: Unexpected error.
+ */
+router.get('/', (req, res, next) => {
```

Comment:
There is no difference between this endpoint and the one above (and those below). You are supposed to have one endpoint that passes req.query to the controller which will then decide what do to _depending_ on those parameters.